infra/tpu-ci/modules/arc-v4-container-cluster/main.tf

```diff
@@ -0,0 +1,97 @@
+provider "google" {
+  project = var.project_id
+}
+
+provider "helm" {
+  kubernetes {
+    host = "https://${google_container_cluster.arc_v4_cluster.endpoint}"
+    token = data.google_client_config.default.access_token
+    cluster_ca_certificate = base64decode(google_container_cluster.arc_v4_cluster.master_auth.0.cluster_ca_certificate)
+  }
+}
+
+data "google_client_config" "default" {}
+
+resource "google_container_cluster" "arc_v4_cluster" {
+  name = var.cluster_name
+  location = "us-central2"
+
+  remove_default_node_pool = true
+  initial_node_count = 1
+
+  release_channel {
+    channel = "RAPID"
+  }
+
+  min_master_version = 1.28
+}
+
+resource "google_container_node_pool" "arc_v4_cpu_nodes" {
+  name = var.cpu_nodepool_name
+  location = "us-central2"
+  cluster = google_container_cluster.arc_v4_cluster.name
+  node_count = var.cpu_node_count
+
+  node_config {
+    oauth_scopes = [
+      "https://www.googleapis.com/auth/logging.write",
+      "https://www.googleapis.com/auth/monitoring",
+    ]
+
+    machine_type = "n1-standard-1"
+  }
+
+  management {
+    auto_upgrade = true
+    auto_repair = true
+  }
+}
+
+resource "google_container_node_pool" "arc_v4_tpu_nodes" {
+  name = var.tpu_nodepool_name
+  location = "us-central2"
+  node_locations = ["us-central2-b"]
+  cluster = google_container_cluster.arc_v4_cluster.name
+  initial_node_count = 0
+  autoscaling {
+    total_min_node_count = 0
```

Comment:
Being able to set a minimum and maximum number of nodes is great! Can you change the minimum to 1 once you start running the workflow? We want at least one warm TPU at a time so we're always able to run at least one job.

Reply:
Will do!